### PR TITLE
Enable Processing of New Golden Pairs

### PR DESCRIPTION
**Pull request by forrestfwilliams**
This PR represents an overhaul of the golden tests for `INSAR_GAMMA` jobs. We are now using 11 new image pairs and have also implemented unique tolerances for each scene and data product.

### CODE REVIEW

**Comment by forrestfwilliams** [comment]
The static analysis is failing due to the import of `rioxarray`. This package is not directly used, but you need to import it to run `xr.open_dataset(path,engine='rasterio')`.

File: tests/test_tifs.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 import hyp3_sdk.util
 import pytest
+import rioxarray
```
Comment (jhkennedy) [inline tests/test_tifs.py]:
```suggestion
import rioxarray  # noqa: F401
```
You can disable Flake8 for this line

https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html

**Comment by forrestfwilliams** [comment]
@jhkennedy I've merged your PR and implemented a fix for the InSAR tests. All the tests pass, but I'm a little suspicious of how fast they're passing and I'm worried some of the code isn't being run. I'm doing a clean re-run of the tests and I'll let you know how it goes.

**Comment by jhkennedy** [comment]
> All the tests pass, but I'm a little suspicious of how fast they're passing and I'm worried some of the code isn't being run.

Sounds like we need to add some logging then! I'll run the InSAR today as well

**Comment by forrestfwilliams** [comment]
The fresh run ran as expected without the issues I mentioned earlier. However, I got unlucky and one of my random hashes was `2021`, which caused the hash replace functionality to replace part of the date for that image. Do we want to control for this, or is this a super-edge case?

**Comment by jhkennedy** [comment]
Wow, we haven't hit that before! I think leaving that as an edge case is just fine.

File: environment.yml [hunk 2]
```diff
@@ -19,3 +19,6 @@ dependencies:
   - rasterio
   - wheel
   - xarray
+  - rioxarray
+  - pip:
+      - remotezip
```
Comment (jhkennedy) [inline environment.yml:24]: I wish this was on conda-forge... maybe we should just do that.

File: hyp3_testing/helpers.py [hunk 3]
```diff
@@ -73,3 +76,37 @@ def clarify_xr_message(message: str, left: str = 'reference', right: str = 'seco
...
+    return product_name, files_normalized
+
+
+@contextmanager
+def job_tifs(job_id, api, directory, keep=False):
+    hyp3 = HyP3(api, os.environ.get('EARTHDATA_LOGIN_USER'), os.environ.get('EARTHDATA_LOGIN_PASSWORD'))
+    job = hyp3.get_job_by_id(job_id)
+
+    product_dir = directory / job.files[0]['filename'].replace('.zip', '')
+    if not product_dir.exists():
```
Comment (jhkennedy) [inline hyp3_testing/helpers.py:101]: This *might* not be robust enough as it's just based on the directory. We could also add a `--force-download` option or something which ensure we re-download everything.

**Review by jhkennedy — APPROVED** [review]
Looks good to me, just a couple thoughts for *future* work.

**Comment by forrestfwilliams** [comment]
Thanks Joe, I also think it might be good to break the golden insar/rtc tests into separate tests for each production/main job pair using a parameterized test. Since data cleanup would happen at the end of every job comparison test, we could avoid some of the complexity currently needed to not download all of the jobs at once.

**Comment by jhkennedy** [comment]
Unfortunately, if you put the setup/tear-down in the parameterization it won't work on a job-by-job basis for PyTest. PyTest does all parameter setup at test collection time, and then does tear-down after all tests have run. So, in our case it'd look like:

* Setup job 1
* Setup job 2
* Setup job 3
...
* run test 1
* run test 2
* run test 3
...
* Tear-down job 3
* Tear-down job 2
* Tear-down job 1

Discussed a bit in these places:
* https://docs.pytest.org/en/7.1.x/example/parametrize.html
* https://stackoverflow.com/questions/54163532/lazy-parametrization-with-pytest
* https://github.com/pytest-dev/pytest/issues/4002
* https://github.com/pytest-dev/pytest/issues/3070

---

I really wanted this to work and spun my wheels on it a bit. Other testing frameworks like `nox` do not have that problem, but I think dumping pytest so we can drop the 2 contex manager lines isn't really worth changing frameworks (esp. having a unique one for this package)

**Pull request merged**